x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/adaptiveallocations/AdaptiveAllocationsScalerService.java

```diff
@@ -75,6 +82,108 @@ Stats sub(Stats value) {
         }
     }
 
+    private class Metrics {
+
+        private final List<AutoCloseable> metrics = new ArrayList<>();
+
+        Metrics() {}
+
+        void init() {
+            if (metrics.isEmpty() == false) {
+                return;
+            }
+            metrics.add(
```

Comment:
Please add a metric for the deployment total inference count which is also in the stats API response.

Reply:
As discussed offline: this is very useful, but not as part of adaptive allocations. Let's implement it somewhere else with different metrics name, etc.